currency-rate/cmd/main.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"time"
+
+	rateProducer "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/broker/rate"
+	userProducer "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/broker/user"
+	appCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/config"
+	cronRate "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/cron/rate"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/models"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/rate/fetchers"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server"
+	serverCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/server/service"
+	transportCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/cron"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database"
+	dbCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/settings"
+)
+
+const (
+	ccFrom = "USD"
+	ccTo   = "UAH"
+)
+
+var appConfig appCfg.Config
+
+const eventTimeout = 5 * time.Second
+
+type UserRepoDecorator struct {
+	*models.UserRepository
+	producer *userProducer.Producer
+}
+
+func (u *UserRepoDecorator) Create(user *models.User) error {
+	if err := u.UserRepository.Create(user); err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), eventTimeout)
+	defer cancel()
+	if err := u.producer.SendSubscribe(ctx, user.Email); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (u *UserRepoDecorator) Delete(user *models.User) error {
+	if err := u.UserRepository.Delete(user); err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), eventTimeout)
+	defer cancel()
+	if err := u.producer.SendUnsubscribe(ctx, user.Email); err != nil {
+		return err
+	}
+	return nil
+}
+
+func InitDatabase() (*database.DB, error) {
```

Comment:
if this is a function constructor, maybe name it as `NewDatabase()` according to the convention?

Reply:
`InitXXX` looks like the built-in [function](https://www.digitalocean.com/community/tutorials/understanding-init-in-go) `init`, which doesn't return anything. In your case it's obviously constructor